backend/gene_info/api/ncbi_provider.py

```diff
@@ -0,0 +1,114 @@
+from ..config import GeneInfoConfig
+import logging
+import urllib.request
+import json
+import xml.etree.ElementTree as ET
+
+
+class NCBIException(Exception):
+    pass
+
+
+class NCBIAPIException(NCBIException):
+    pass
+
+
+class NCBIUnexpectedResultException(NCBIException):
+    pass
+
+
+class NCBIProvider(object):
+    """
+    Provider class used to generate NCBI URL
+    """
+
+    def __init__(self) -> None:
+        self.base_ncbi_uri = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/"
+        gene_info_config = GeneInfoConfig()
+        gene_info_config.load()
+        try:
+            self.ncbi_api_key = f"&api_key={gene_info_config.ncbi_api_key}"
+        except RuntimeError:
+            logging.error("Could not find NCBI API key")
+            self.ncbi_api_key = None
+        super().__init__()
+
+    def _get_api_key(self):
```

Comment:
nit: I think this should be a [property](https://docs.python.org/3/library/functions.html#property) or a class instance variable and only resolved once.